llvm/include/llvm/ADT/ADL.h

```diff
@@ -9,6 +9,9 @@
 #ifndef LLVM_ADT_ADL_H
 #define LLVM_ADT_ADL_H
 
+#if defined(_MSC_VER) && _MSC_VER < 1937
```

Comment:
@compnerd You suggested checking for the following to me but that check did not evaluate to `true` in Windows CI.

```cpp
#if defined(_MSVC_STL_UPDATE) && _MSVC_STL_UPDATE < 202305l
```

Based on the `_MSVC_STL_UPDATE` <-> `_MSC_VER` mapping, I changed it to a check for `_MSC_VER`. Does this make sense to you?

Reply:
That doesn't really work.  `_MSC_VER` is the compiler version, the problem here is the header version, which is `_MSVC_STL_UPDATE`.  I think that we should just figure out what version is in use and use that.

Reply:
Do you have any suggestions I should try?